spinalcordtoolbox/math.py

```diff
@@ -113,14 +106,61 @@ def dilate(data, size, shape, dim=None):
     """
     if isinstance(data, Image):
         im_out = data.copy()
-        im_out.data = dilate(data.data, size, shape, dim)
+        im_out.data = dilate(data.data, size, shape, dim, islabel)
         return im_out
     else:
         footprint = _get_footprint(shape, size, dim)
-        if data.dtype in ['uint8', 'uint16']:
-            return rank.maximum(data, footprint=footprint)
+        if islabel:
+            return _dilate_point_labels(data, footprint=footprint)
         else:
-            return dilation(data, footprint=footprint, out=None)
+            if data.dtype in ['uint8', 'uint16']:
+                return rank.maximum(data, footprint=footprint)
+            else:
+                return dilation(data, footprint=footprint)
+
+
+def _dilate_point_labels(data, footprint):
+    """
+    A more efficient dilation algorithm when we know the image is mostly zero (i.e. point label image)
+    """
+    if len(data.shape) != len(footprint.shape):
+        raise ValueError(f"incompatible shapes: {data.shape=} {footprint.shape=}")
+    if footprint.size == 0:
+        raise ValueError(f"cannot handle empty footprint: {footprint.shape=}")
+
+    data_shape = np.array(data.shape)
+    fp_shape = np.array(footprint.shape)
+
+    # For odd-length dimensions, the center of the footprint is unique.
+    # For even-length dimensions, we round down.
+    fp_center = (fp_shape - 1) // 2
+
+    data_out = np.zeros_like(data)
+    for data_coords in np.argwhere(data):
+        # data_coords is a numpy array, which behaves differently from tuples
+        # when indexing another numpy array. We want the tuple behaviour.
+        data_value = data[tuple(data_coords)]
+
+        # If a data coordinate is too close to the edge, we have to use a
+        # truncated version of the footprint. These formulas compute the
+        # adjusted slice start and stop coordinates in the data array, and in
+        # the footprint array.
+        pre_distances = np.min(data_coords, fp_center)
```

Comment:
The error seems to be: 

```
   File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/math.py", line 148, in _dilate_point_labels
    pre_distances = np.min(data_coords, fp_center)
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 2953, in min
    return _wrapreduction(a, np.minimum, 'min', axis, None, out,
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 88, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
TypeError: only integer scalar arrays can be converted to a scalar index
```

From SO:

> I ran into the problem when venturing to use numpy.concatenate to emulate a C++ like pushback for 2D-vectors; If A and B are two 2D numpy.arrays, then numpy.concatenate(A,B) yields the error.
> The fix was to simply to add the missing brackets: numpy.concatenate( ( A,B ) ), which are required because the arrays to be concatenated constitute to a single argument

So it looks like the fix here is just:

```suggestion
        pre_distances = np.min((data_coords, fp_center))
```